### PR TITLE
Added log formats and support methods.

### PR DESCRIPTION
**Pull request by Skandalik**

### CODE REVIEW

File: logger.go [hunk 1]
```diff
@@ -55,6 +56,39 @@ func (l Level) String() string {
...
+const (
+	Json Format = iota
+	Logfmt
+)
+
+// Format represents the predefined log format.
+type Format int
+
+// formatFromString returns a formatter instance appropriate for the given format name.
+func FormatFromString(format string) (Formatter, error) {
```
Comment (boguslaw-wojcik) [inline logger.go]: FormatterFromString

Comment (boguslaw-wojcik) [reply logger.go]: FormatFromString would return a const.

File: logger.go [hunk 2]
```diff
@@ -55,6 +56,39 @@ func (l Level) String() string {
...
+		return LogfmtFormat(), nil
+	default:
+		return nil, xerrors.Errorf("log: invalid log format: %s", format)
+	}
+}
+
+// String returns the string representation of the level.
+func (f Format) String() string {
+	switch f {
+	case Json:
```
Comment (boguslaw-wojcik) [inline logger.go]: Not sure if this should not be JSON instead of Json

File: logger.go (hunk 1 above)
Comment (Skandalik) [reply logger.go]: Good catch. Fixed.

File: logger.go (hunk 2 above)
Comment (Skandalik) [reply logger.go]: Going with the Go convention, it should be JSON I guess. Will fix it.

**Review by boguslaw-wojcik — APPROVED** [review]

**Pull request merged**